### ISSUE TITLE
Not warning about tapping an empty hash

### ISSUE THREAD

**Issue by guyisra** [bbatsov/rubocop#3075]
code that is doing

```
{}.tap do |hash|
hash[:key] = "value"
end
```

is pointless, it should warn and just convert it to 
{key: value}

**Comment by Drenmi** [comment]
Hm. As a quick note, this is only pointless as long as none of the assignments are conditional. Consider:

```
{}.tap do |hash|
  hash[:one] = :two
  hash[:three] = :four if something?
end
```

**Comment by alexdowad** [comment]
You are right that the code is pointless. But I have never seen such code ever used in the "real world".

I would suggest that this problem is not common enough to justify having a dedicated cop in RC core.

**Comment by bbatsov** [comment]
Yeah, seems a rather uncommon problem, so I'd rather us not implement a cop for this.
